### PR TITLE
Suggest types (and intrinsic type keywords) after "(" when declaring …

### LINKED ISSUE #14127: Completion doesn't show while typing tuple return types
**Version Used**: preview/25725

```C#
class C
{
    ($$)
}
```
I'd expect completion to work here, since I'm trying to find a type name

### PR DESCRIPTION
**Pull request by rchande**
…a tuple type as the type of a member.

Fixes #14127

### CODE REVIEW

**Comment by rchande** [comment]
Tag @dotnet/roslyn-ide @jcouv for review

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs [hunk 1]
```diff
@@ -8890,5 +8890,49 @@ void foo()
...
+        }
+
+        [WorkItem(14127, "https://github.com/dotnet/roslyn/issues/14127")]
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async Task TupleTypeAtMemberLevel3()
+        {
+            await VerifyItemExistsAsync(@"
+class C
+{
+    (C,
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs]: I'm confused by the lack of $$'s in this test...

File: src/EditorFeatures/CSharpTest2/Recommendations/SByteKeywordRecommenderTests.cs [hunk 2]
```diff
@@ -623,5 +623,30 @@ static void Main(string[] args)
...
+        public async Task TestInTupleWithinMember()
+        {
+            await VerifyKeywordAsync(@"
+class Program
+{
+    void Method()
+    {
+        ($$
+    }
+}");
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest2/Recommendations/SByteKeywordRecommenderTests.cs]: Also need tests for compound foreach.  Also, at least a test or so for tuples in other areas (parameter lists, etc.)

File: src/Features/CSharp/Portable/Completion/KeywordRecommenders/StringKeywordRecommender.cs [hunk 3]
```diff
@@ -36,7 +36,9 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
                 syntaxTree.IsDefaultExpressionContext(position, context.LeftToken, cancellationToken) ||
                 syntaxTree.IsAfterKeyword(position, SyntaxKind.ConstKeyword, cancellationToken) ||
                 context.IsDelegateReturnTypeContext ||
+                context.IsPossibleTupleContext ||
                 syntaxTree.IsGlobalMemberDeclarationContext(position, SyntaxKindSet.AllGlobalMemberModifiers, cancellationToken) ||
+                context.IsPossibleTupleContext ||
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Completion/KeywordRecommenders/StringKeywordRecommender.cs]: whoops

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs [hunk 4]
```diff
@@ -1139,10 +1140,10 @@ public static bool IsPossibleTupleElementNameContext(this SyntaxToken token, int
 
             if (token.IsKind(SyntaxKind.OpenParenToken))
             {
-                return token.Parent.IsKind(SyntaxKind.ParenthesizedExpression) || token.Parent.IsKind(SyntaxKind.TupleExpression);
+                return token.Parent.IsKind(SyntaxKind.ParenthesizedExpression) || token.Parent.IsKind(SyntaxKind.TupleExpression, SyntaxKind.TupleType);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs]: this doesn't make sense given the method you're in.  If you're after the open paren in a tuple-type then that is  *not* a "PossibleTupleElementNameContext".  Sounds like you might need an IsTupleElementTypeContext or IsPossibleTupleElementTypeContext or something.

Comment (jcouv) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs]: It's possible this method is misnamed. It has already been used for more than just tuple element name context.
Maybe rename to `IsPossibleTupleContext`?

File: src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs (hunk 1 above)
Comment (rchande) [inline src/EditorFeatures/CSharpTest/Completion/CompletionProviders/SymbolCompletionProviderTests.cs]: Fixed

File: src/EditorFeatures/CSharpTest2/Recommendations/SByteKeywordRecommenderTests.cs (hunk 2 above)
Comment (rchande) [inline src/EditorFeatures/CSharpTest2/Recommendations/SByteKeywordRecommenderTests.cs]: Done

File: src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs (hunk 4 above)
Comment (rchande) [inline src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs]: Sounds good @jcouv -- I've renamed the method to reflect the more general usage.

**Comment by rchande** [comment]
@333fred My unit test runs are failing with

```
There are errors when processing test results

There was an error processing the xunit test results: [xUnit] [ERROR] - No test reports found for the metric 'xUnit.Net' with the resolved pattern '**/xUnitResults/*.xml'. Configuration error?.
Indication 1 
```

Know issue? Can you take a look?

**Comment by 333fred** [comment]
@TyOverby @jaredpar 

I think this is another case of the cmd escaping in the perf tests failing.

```
17:00:28 SUCCESS: The process "VBCSCompiler.exe" with PID 3008 has been terminated.
17:00:28 after was unexpected at this time.
```

**Comment by rchande** [comment]
retest this please

**Comment by rchande** [comment]
test this please

**Comment by rchande** [comment]
retest windows_debug_unit32_prtest please

**Comment by 333fred** [comment]
@rchande, this is because #14480. For a quick workaround, you can rename your pull.

**Comment by rchande** [comment]
retest this please

**Comment by rchande** [comment]
retest this please

**Pull request merged**